api/v1alpha1/templatechain_common.go

```diff
@@ -14,12 +14,25 @@
 
 package v1alpha1
 
-// TemplateChainSpec defines the observed state of TemplateChain
+import (
+	"fmt"
+	"slices"
+)
+
+// TemplateChainSpec defines the desired state of *TemplateChain
 type TemplateChainSpec struct {
 	// SupportedTemplates is the list of supported Templates definitions and all available upgrade sequences for it.
 	SupportedTemplates []SupportedTemplate `json:"supportedTemplates,omitempty"`
 }
 
+// TemplateChainSpec defines the desired state of *TemplateChain
```

Comment:
```suggestion
// TemplateChainStatus defines the observed state of *TemplateChain
```
or something like this

Reply:
amended, thanks for noticing